components/announcement_bar/text_dismissable_bar.tsx

```diff
@@ -13,27 +12,31 @@ import AnnouncementBar from './default_announcement_bar';
 
 const localStoragePrefix = '__announcement__';
 
-export default class TextDismissableBar extends React.PureComponent {
-    static propTypes = {
-        allowDismissal: PropTypes.bool.isRequired,
-        text: PropTypes.string.isRequired,
-        onDismissal: PropTypes.func,
+type AnnouncementBarProps = React.ComponentProps<typeof AnnouncementBar>;
 
-        // Any props that AnnouncementBar supports
-    };
+interface Props extends Partial<AnnouncementBarProps> {
+    allowDismissal?: boolean;
+    text?: string;
```

Comment:
Should `allowDismissal` and `text` be optional? It seems that in the original props they were required